tests/utils.py

```diff
@@ -169,12 +169,17 @@ def get_image_name_from_csv(image_string, csv_related_images):
     raise ResourceNotFoundError(f"no image with the string {image_string} was found in the csv_dict")
 
 
-def hotplug_resource_and_wait_hotplug_migration_finish(vm, client, sockets=None, memory_guest=None):
-    assert sockets or memory_guest, "No resource for update provided!!!"
-    assert not (sockets and memory_guest), "Hotpluging both cpu and memory at once is not supported!"
+def hotplug_resource_and_wait_hotplug_migration_finish(
+    vm, client, sockets=None, memory_guest=None, new_instance_type=None
+):
+    assert bool(sockets or memory_guest) ^ bool(new_instance_type), "wrong parameter values for Hotpluging"
+    if new_instance_type:
+        instance_type_spec = new_instance_type.instance.spec
+        sockets = instance_type_spec.cpu.guest
+        memory_guest = instance_type_spec.memory.guest
 
     if vm.instance.spec.get("instancetype"):
-        hotplug_instance_type_vm(vm=vm, sockets=sockets, memory_guest=memory_guest)
+        hotplug_instance_type_vm(vm=vm, new_instance_type_name=new_instance_type.name)
```

Comment:
is there a needcheck for a RestartRequired condition after hotplugging and before migrating

Reply:
The restart should not be needed at all, one you change the name, it will migrate and change the VMI values.

Do you mean should we check a `RestartRequired` message does not show?

Reply:
https://issues.redhat.com/browse/CNV-55743  --> large-->small - No change in cpu/memory until we restart , so basically nothing happens

Reply:
i mean outside if-else block?

Reply:
There's a specific test to check for the `RestartRequired` message -> test_decrease_cpu_value:
This test checks for the alert raising: "Reduction of CPU socket count requires a restart".

Is this what you meant? if so I don't think we should check the alert on all hot-plugs, just the ones we expect it.

Reply:
I suspect a bug there. Let's check

Reply:
https://issues.redhat.com/browse/CNV-60547